bpf/lib/lb.h

```diff
@@ -449,11 +450,32 @@ static __always_inline int __lb6_rev_nat(struct __ctx_buff *ctx, int l4_off,
 		new_saddr = tmp.addr;
 	}
 
+	if (ct_state->loopback) {
+		/* The packet was looped back to the sending endpoint on the
+		 * forward service translation.
+		 */
+		union v6addr old_daddr;
+
+		if (ipv6_load_daddr(ctx, ETH_HLEN, &old_daddr) < 0)
+			return DROP_INVALID;
```

Comment:
I am not entirely sure if `tuple->daddr` will have the required destination address. I duplicated the logic from IPv4 rev nat handling here - https://github.com/cilium/cilium/blob/d50075de642c913e59e65a58201edf7337420dba/bpf/lib/lb.h#L944-L967
Looking at the code we do something similar for old_saddr where it will either loaded from the context or the tuple depending on the flags, so my guess is that tuple->daddr can also be empty. https://github.com/cilium/cilium/pull/14138/commits/69ff2f330f1c9aae375c2b64c9f5ff16358b4dbd#diff-91f8d692a44213031badf4fb6cc4394028599b992e4f847d99fc4542ac8f05f8R441-R451